src/main/java/seedu/ccacommander/logic/commands/UnenrolCommand.java

```diff
@@ -0,0 +1,127 @@
+        List<Enrolment> lastShownEnrolmentList = model.getFilteredEnrolmentList();
+        List<Member> lastShownMemberList = model.getFilteredMemberList();
+        List<Event> lastShownEventList = model.getFilteredEventList();
+
+        int zeroBasedMemberIndex = memberIndex.getZeroBased();
+        int zeroBasedEventIndex = eventIndex.getZeroBased();
+
+        if (zeroBasedMemberIndex >= lastShownMemberList.size()
+                && zeroBasedEventIndex < lastShownEventList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_MEMBER_DISPLAYED_INDEX);
+        } else if (zeroBasedEventIndex >= lastShownEventList.size()
+                && zeroBasedMemberIndex < lastShownMemberList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_EVENT_DISPLAYED_INDEX);
+        } else if (zeroBasedEventIndex >= lastShownMemberList.size()
+                && zeroBasedEventIndex >= lastShownEventList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_MEMBER_AND_EVENT_DISPLAYED_INDEX);
+        }
+
+        Member member = lastShownMemberList.get(zeroBasedMemberIndex);
+        Event event = lastShownEventList.get(zeroBasedEventIndex);
+
+        Name memberName = member.getName();
+        Name eventName = event.getName();
+
+        Enrolment enrolmentToDelete = findEnrolmentFromList(lastShownEnrolmentList, memberName, eventName);
+        model.deleteEnrolment(enrolmentToDelete);
+        model.commit(String.format(MESSAGE_COMMIT, enrolmentToDelete.getMemberAndEventEnrolment()));
+        return new CommandResult(String.format(MESSAGE_DELETE_ENROLMENT_SUCCESS, Messages.format(enrolmentToDelete)));
+    }
+
+    /**
+     *  Checks if the {@code lastShownEnrolmentList} contains a specific Enrolment object that has a Member and an
+     *  Event that matches the given {@param memberName} and {@param eventName} respectively, and returns that
+     *  specific Enrolment.
+     * @param lastShownEnrolmentList
+     * @param memberName
+     * @param eventName
+     * @throws EnrolmentNotFoundException if the enrolment cannot be found from the {@code lastShownEnrolmentList}
+     */
+    public static Enrolment findEnrolmentFromList(List<Enrolment> lastShownEnrolmentList,
+        Name memberName, Name eventName) throws EnrolmentNotFoundException {
+        Enrolment selectedEnrolment = null;
+
+        for (int i = 0; i < lastShownEnrolmentList.size(); i++) {
+            selectedEnrolment = lastShownEnrolmentList.get(i);
+            Name selectedMemberName = selectedEnrolment.getMemberName();
+            Name selectedEventName = selectedEnrolment.getEventName();
+
+            if (memberName.equals(selectedMemberName) && eventName.equals(selectedEventName)) {
+                break;
+            }
+        }
+
+        if (selectedEnrolment == null) {
+            throw new EnrolmentNotFoundException();
+        } else {
+            return selectedEnrolment;
+        }
```

Comment:
Would it make sense to move this function elsewhere maybe to Model since this seems quite reusable.

Reply:
Good suggestion, will move this to the `ModelManager` class.